sqlair.go

```diff
@@ -163,3 +164,61 @@ func (q *Query) One(outputArgs ...any) error {
 	iter.Decode(outputArgs...)
 	return iter.Close()
 }
+
+// All iterates over the query and decodes all the rows.
+// resultColumns must contain pointers to slices of each of the query arguments e.g.
+// > q.All(&[]*Person{}, &[]*Address{}) or q.All(&[]Address{}, &[]Person{})
+// It fabricates all structs as needed.
+func (q *Query) All(slicePointers ...any) error {
+	if q.err != nil {
+		return q.err
+	}
+
+	// Check slice inputs
+	var slicePtrVals = []reflect.Value{}
+	var sliceVals = []reflect.Value{}
+	for _, ptr := range slicePointers {
+		ptrVal := reflect.ValueOf(ptr)
+		if ptrVal.Kind() != reflect.Pointer {
+			return fmt.Errorf("need pointer to slice, got %s", ptrVal.Kind())
+		}
+		if ptrVal.IsNil() {
+			return fmt.Errorf("need pointer to slice, got nil")
+		}
+		slicePtrVals = append(slicePtrVals, ptrVal)
+		sliceVal := ptrVal.Elem()
+		if sliceVal.Kind() != reflect.Slice {
+			return fmt.Errorf("need pointer to slice, got pointer to %s", sliceVal.Kind())
+		}
+		sliceVals = append(sliceVals, sliceVal)
+	}
+
+	iter := q.Iter()
+	for iter.Next() {
+		var outputArgs = []any{}
+		for _, sliceVal := range sliceVals {
+			elemType := sliceVal.Type().Elem()
+			if elemType.Kind() != reflect.Pointer {
+				return fmt.Errorf("need pointer to slice of pointers, got pointer to slice of %s", elemType.Kind())
```

Comment:
This is a bit unclear, but more fundamentally, should this even be an error?  If we get `[]People`, that's not a slice of pointers, and we do want to support it as it's significantly more efficient.

Reply:
Changed.